Problem: releases do not include files required for cmake

# Issue description

libzmq releases do not include files required to build with cmake.

# Environment

* libzmq version (commit hash if unreleased): c8a1c4542d13b6492949e7525f4fe8da266cac2b

# Minimal test code / Steps to reproduce the issue

```
./configure
make dist
tar -xzf zeromq-4.3.1.tar.gz
cd zeromq-4.3.1
mkdir build
cd build
cmake ..
```

# What's the actual result?

```
...
-- Found AsciiDoc: /usr/local/bin/asciidoc
CMake Warning (dev) at tests/CMakeLists.txt:239 (message):
  Test 'test_security_gssapi' is not known to CTest.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error: File ~/dev/zmq/libzmq/zeromq-4.3.1/builds/cmake/clang-format-check.sh.in does not exist.
CMake Error at builds/cmake/Modules/ClangFormat.cmake:27 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:1295 (include)
```

# What's the expected result?

Start building zeromq


# Proposed fix:

To protect against these issues, the tests should probably be run from releases of zeromq after `make dist`, rather than directly from the repo.

--- a/Makefile.am
+++ b/Makefile.am
@@ -944,6 +944,7 @@ EXTRA_DIST = \
 	autogen.sh	\
 	version.sh	\
 	src/libzmq.pc.cmake.in \
+	ci_build.sh \
 	src/libzmq.vers \
 	src/version.rc.in \
 	tests/CMakeLists.txt \

--- a/builds/Makefile.am
+++ b/builds/Makefile.am
@@ -15,15 +15,20 @@ EXTRA_DIST = \
 	cygwin/Makefile.cygwin \
 	mingw32/Makefile.mingw32 \
 	mingw32/platform.hpp \
+	cmake/ci_build.sh \
 	cmake/Modules \
 	cmake/NSIS.template32.in \
 	cmake/NSIS.template64.in \
 	cmake/ZeroMQConfig.cmake.in \
+	cmake/clang-format-check.sh.in \
 	cmake/platform.hpp.in \
+	valgrind/ci_build.sh \
 	valgrind/valgrind.supp \
 	valgrind/vg \
 	nuget/readme.nuget \
 	nuget/libzmq.autopkg \
+	android/Dockerfile \
+	android/README.md \
 	android/android_build_helper.sh \
 	android/ci_build.sh \
 	android/build.sh

--- a/ci_build.sh
+++ b/ci_build.sh
@@ -3,6 +3,16 @@
 set -x
 set -e
 
+# always run tests from dist
+# to make sure that `make dist` doesn't omit any files required for the tests
+./autogen.sh
+./configure
+make -j5 dist-gzip
+V=$(./version.sh)
+tar -xzf zeromq-$V.tar.gz
+cd zeromq-$V
+
+
 if [ $BUILD_TYPE == "default" ]; then
     mkdir tmp
     BUILD_PREFIX=$PWD/tmp

--- a/ci_build.sh
+++ b/ci_build.sh
@@ -70,14 +70,17 @@ if [ $BUILD_TYPE == "default" ]; then
         make VERBOSE=1 -j5 distcheck
     ) || exit 1
 else
-    # always install from dist
+    # always install custom builds from dist
     # to make sure that `make dist` doesn't omit any files required to build & test
-    ./autogen.sh
-    ./configure
-    make -j5 dist-gzip
-    V=$(./version.sh)
-    tar -xzf zeromq-$V.tar.gz
-    cd zeromq-$V
+    # coverage, valgrind are special-case tests that aren't meant to be runnable from releases
+    if [ ${BUILD_TYPE} != "coverage" ] && [ ${BUILD_TYPE} != "valgrind" ]; then
+        ./autogen.sh
+        ./configure
+        make -j5 dist-gzip
+        V=$(./version.sh)
+        tar -xzf zeromq-$V.tar.gz
+        cd zeromq-$V
+    fi
 
     # start the actual build
     cd ./builds/${BUILD_TYPE} && ./ci_build.sh

--- a/ci_build.sh
+++ b/ci_build.sh
@@ -3,16 +3,6 @@
 set -x
 set -e
 
-# always run tests from dist
-# to make sure that `make dist` doesn't omit any files required for the tests
-./autogen.sh
-./configure
-make -j5 dist-gzip
-V=$(./version.sh)
-tar -xzf zeromq-$V.tar.gz
-cd zeromq-$V
-
-
 if [ $BUILD_TYPE == "default" ]; then
     mkdir tmp
     BUILD_PREFIX=$PWD/tmp
@@ -80,5 +70,15 @@ if [ $BUILD_TYPE == "default" ]; then
         make VERBOSE=1 -j5 distcheck
     ) || exit 1
 else
+    # always install from dist
+    # to make sure that `make dist` doesn't omit any files required to build & test
+    ./autogen.sh
+    ./configure
+    make -j5 dist-gzip
+    V=$(./version.sh)
+    tar -xzf zeromq-$V.tar.gz
+    cd zeromq-$V
+
+    # start the actual build
     cd ./builds/${BUILD_TYPE} && ./ci_build.sh
 fi

--- a/ci_build.sh
+++ b/ci_build.sh
@@ -82,6 +82,6 @@ else
         cd zeromq-$V
     fi
 
-    # start the actual build
+    # start the actual build from inside the dist
     cd ./builds/${BUILD_TYPE} && ./ci_build.sh
 fi

